### ISSUE TITLE
Make a WASM / WebR compatible version of the package

### ISSUE THREAD

**Issue by danm0nster** [morenococo/crqa#8]
Several dependencies prevent the package in its current state from compiling to WASM and working with WebR.

It would be great to have a version that enables this without breaking backwards compatibility.

The problematic parts are:

* `Plot3D` package (not available for WebR and won't be, since it uses Tcl/Tk)
* `lorenzattractor()` function
* `tseriesChaos` package (not available for WebR)
* `gplots` package (not available for WebR)

I have [a working fork](https://github.com/danm0nster/crqa) of this repository that can be cross compiled to WASM and works in WebR.

An example of how it can be used can be found [here](https://tildeweb.au.dk/~au78495/NLM/interactive.html).

**Comment by morenococo** [comment]
very nice and extremely useful. I would say that if the changes do not disrupt any of the core functions, i.e., those that have directly to do with computing RQA measures, then that's ok. Plot3D was just there for illustration and the lorenzattractor() could work without it, and should not require anything else than base R. 
Not sure instead about tseriesChaos and gplots, whether they are accessed in other more substantial sections of the code. I have assigned you and me to double check this.

**Comment by morenococo** [comment]
@danm0nster I did some work on this. I do not understand why tseriesChaos gives you an error if the function you rewrote in your fork still calls a .C function within tseriesChaos. What I mean is that the package is still called by WebR

**Comment by danm0nster** [comment]
@morenococo, what is the compiled C function that is called? I did not notice that, and the compilation for WebR worked fine.

Further regarding the WebR issue, there were several package dependencies that I needed to remove. I simply dropped in two functions from tseriesChaos, and I am not sure that is kosher. It was just a quick-and-dirty fix, to get it running as a proof-of-concept. The other thing I did was replace the call to gplots::hist2d()  with some code that I did not check at all, so it probably does not do the right thing. Again: a quick-and-dirty fix.

So to get webR support, we probably need to invest some more time and do some testing.

**Comment by morenococo** [comment]
@danm0nster yes i saw your fixes and tried to implement them into a package that compile for CRAN but, take for example, line 8 of false.nearest from your fork

a <- .C("falseNearest", series=as.double(x), length=as.integer(length(series)), m=as.integer(i), d=as.integer(d), t=as.integer(t), eps=as.double(eps), rt=as.double(rt), out=as.double(res[i]), out2=as.integer(res2[i]), PACKAGE="tseriesChaos")

it is calling a .C function from tseriesChaos, so the weird thing is that this does not fix the dependency to the package, and so, if I compile for our crqa() I still need to state the dependency to it. So, i am a bit puzzled about what exactly is creating a conflict with WebR. The same goes for ami.  
hist2d could instead be ported from gplots as an independent function, and we can add it to the package because it does not have this dependency issue.

**Comment by danm0nster** [comment]
It is not possible to install `tseriesChaos` in webR, because it depends on `deSolve` which is not available for webR. 

This is what happens if I try in a running instance of webR:

```
> webr::install("tseriesChaos")
Warning in install(dep, repos, info, lib, quiet, mount) :
  Requested package deSolve not found in webR binary repo.
Downloading webR package: tseriesChaos
```

I also tried using the shim install method:

```
> webr::shim_install()
> install.packages("tseriesChaos")
Warning in install(dep, repos, info, lib, quiet, mount) :
  Requested package deSolve not found in webR binary repo.
```
You can test this using the [demo of the webR REPL](https://webr.r-wasm.org/latest/).

That said, I don't understand how I was able to compile a webR version of `crqa` without the C-files. That remains a mystery to me, and think I will need to reproduce alll the steps needed to cross compile one more time to figure it out.

**Comment by danm0nster** [comment]
Just a thought: since the `.C()` call has the parameter `PACKAGE="tseriesChaos"` then --- maybe --- the compilation environment is smart enough to grab the file `false.nearest.c` with the function `falseNearest` from the source code of the `tseriesChaos` package. That does not seem entirely unlikely to me, and it would explain why it works. I would have to redo the cross compilation and look at the output in order to verify this, of course.  I can't seem to find any log files from the original compilation.
